opentelemetry-otlp/src/proto.rs

```diff
@@ -1,5 +1,5 @@
 #[cfg(feature = "tonic")]
-pub(crate) mod collector {
+pub mod collector {
```

Comment:
An alternative could be just to have the integration in test in `src/tests.rs`, do you think that could work? It would prevent having to have the `integraion-testing` feature and all the conditionally public APIs.

Reply:
yeah, i've been thinking about this a bit.

the part that i really don't like is the change i need to apply to the build script. I would love to just have my own build script for the integration test which would make all of this go away, since i would be able to generate the grpc server myself.

since we don't want to build the gRPC server when not in the integration-test i don't really see a good way to get rid of the feature flag though.

Reply:
We could just have a separate crate for the integration test, right?

Reply:
Couldn't we use `cfg(test)` to conditionally build the gRPC server only for test scenarios?

Reply:
oh, good point.

any ideas how a separate crate would integrate with the cargo test runner?

Reply:
Just have a separate crate with an integration test in it?

Reply:
i explored the unit test idea, but looks like `cfg!(test)` does not work in `build.rs` - there are some open cargo issues about this.

Reply:
is the separate crate what you'd prefer over my current solution?

Reply:
Actually a separate crate would probably not solve it, it'd still need the internal APIs exposed to the test crate. So that probably doesn't make sense, then.

Reply:
it would work, b/c i would have a separate build script using the same proto files. i don't think it would need any special modules